label_sleuth/orchestrator/orchestrator_api.py

```diff
@@ -322,7 +322,10 @@ def delete_iteration_model(self, workspace_id, category_name, iteration_index):
     def _delete_category_models(self, workspace_id, category_name):
         workspace = self.orchestrator_state.get_workspace(workspace_id)
         for idx in range(len(workspace.categories[category_name].iterations)):
-            self.delete_iteration_model(workspace_id, category_name, idx)
+            try:
+                self.delete_iteration_model(workspace_id, category_name, idx)
+            except Exception as e:
+                logging.exception(f"error deleting model for iteration '{idx}' in category '{category_name}' from workspace '{workspace_id}'")
```

Comment:
this is a working solution, but a better solution would be to check for the model status and only try to delete it if the status is not deleted.
you can get the model status using `workspace.categories[category_id].iterations[idx].model.model_status`
Thanks!